packages/devtools_app/lib/src/screens/memory/panes/profile/profile_view.dart

```diff
@@ -204,6 +205,209 @@ class _FieldSizeColumn extends ColumnData<ProfileRecord> {
+        return profile.newSpaceGCStats;
+      case _HeapGeneration.oldSpace:
+        return profile.oldSpaceGCStats;
+      case _HeapGeneration.total:
+        return profile.totalGCStats;
+    }
+  }
+}
+
+class _HeapNameColumn extends ColumnData<AdaptedProfile> {
+  _HeapNameColumn()
+      : super(
+          '',
+          fixedWidthPx: scaleByFontFactor(200),
+        );
+
+  @override
+  String? getValue(AdaptedProfile dataObject) {
+    return 'GC Statistics';
+  }
+}
+
+class _HeapUsageColumn extends _HeapStatsColumn {
+  _HeapUsageColumn({required super.generation})
+      : super(
+          'Usage',
+          titleTooltip: 'The current amount of memory allocated from the heap',
+          alignment: ColumnAlignment.right,
+          fixedWidthPx: scaleByFontFactor(_defaultNumberFieldWidth),
+        );
+
+  @override
+  int? getValue(AdaptedProfile dataObject) {
+    return getGCStats(dataObject).usage;
+  }
+
+  @override
+  String getDisplayValue(AdaptedProfile dataObject) {
+    return prettyPrintBytes(getValue(dataObject), includeUnit: true)!;
+  }
+
+  @override
+  bool get numeric => true;
+}
+
+class _HeapCapacityColumn extends _HeapStatsColumn {
+  _HeapCapacityColumn({required super.generation})
+      : super(
+          'Capacity',
+          titleTooltip:
+              'The current size of the heap, including unallocated memory',
+          alignment: ColumnAlignment.right,
+          fixedWidthPx: scaleByFontFactor(_defaultNumberFieldWidth),
+        );
+
+  @override
+  int? getValue(AdaptedProfile dataObject) {
+    return getGCStats(dataObject).capacity;
+  }
+
+  @override
+  String getDisplayValue(AdaptedProfile dataObject) {
+    return prettyPrintBytes(getValue(dataObject), includeUnit: true)!;
+  }
+
+  @override
+  bool get numeric => true;
+}
+
+class _HeapCollectionsColumn extends _HeapStatsColumn {
+  _HeapCollectionsColumn({required super.generation})
+      : super(
+          'Collections',
+          titleTooltip: 'The number of garbage collections run on the heap',
+          alignment: ColumnAlignment.right,
+          fixedWidthPx: scaleByFontFactor(_defaultNumberFieldWidth),
+        );
+
+  @override
+  int? getValue(AdaptedProfile dataObject) {
+    return getGCStats(dataObject).collections;
+  }
+
+  @override
+  bool get numeric => true;
+}
+
+class _HeapCollectionsTime extends _HeapStatsColumn {
+  _HeapCollectionsTime({required super.generation})
+      : super(
+          'Latency',
+          titleTooltip:
+              'The average time taken to perform a garbage collection on the heap (ms)',
+          alignment: ColumnAlignment.right,
+          fixedWidthPx: scaleByFontFactor(_defaultNumberFieldWidth),
+        );
+
+  @override
+  double? getValue(AdaptedProfile dataObject) {
+    final stats = getGCStats(dataObject);
+    if (stats.averageCollectionTime.isNaN) {
+      return 0;
+    }
+    return getGCStats(dataObject).averageCollectionTime * 1000;
```

Comment:
we should use the msText helper here

Reply:
This isn't returning text and `averageCollectionTime` is in seconds.

Reply:
We use `msText` below.